### PR TITLE
[Serve] Make `RandomKiller` restart in `long_running_serve_failure` test

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
Signed-off-by: Shreyas Krishnaswamy <shrekris@anyscale.com>

## Why are these changes needed?

The `long_running_serve_failure` test uses a long-running actor, `RandomKiller`, to randomly kill Serve actors. This change sets the `RandomKiller`'s `max_restarts` and `max_task_retries` to `-1`, so it can restart after crashes.

## Related issue number

Addresses #31741

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [X] Release tests
     - This change updates the `long_running_serve_failure` release test.

### CODE REVIEW

**Review by sihanwang41 — APPROVED** [review]

**Review by architkulkarni — APPROVED** [review]
Nice catch! ~How did you find out that the RandomKiller was crashing?~ (Ah I see the logs you sent offline)

File: release/long_running_tests/workloads/serve_failure.py [hunk 1]
```diff
@@ -172,5 +172,6 @@ def run(self):
                 break
 
 
-tester = RandomTest(max_deployments=NUM_NODES * CPUS_PER_NODE)
+random_killer = RandomKiller.remote()
```
Comment (architkulkarni) [inline release/long_running_tests/workloads/serve_failure.py:175]: What's the effect of pulling it out of RandomTest?

**Comment by architkulkarni** [comment]
Will merge as soon as CI passes (i.e. without blocking on running the release test in this PR) since the test is already flaky and we're monitoring it closely. That way we can get more runs in to verify the flakiness is fixed before the branch cut.

**Comment by shrekris-anyscale** [comment]
> Nice catch! How did you find out that the RandomKiller was crashing?

I kicked off [the test yesterday](https://github.com/ray-project/ray/issues/31741#issuecomment-1405658465), but it failed after ~10 hours with the error:

```
Traceback (most recent call last):
  File "workloads/serve_failure.py", line 176, in <module>
    tester.run()
  File "workloads/serve_failure.py", line 149, in run
    action_chosen()
  File "workloads/serve_failure.py", line 128, in create_deployment
    ray.get(self.random_killer.stop_spare.remote(new_name))
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/worker.py", line 2384, in get
    raise value
ray.exceptions.RayActorError: The actor died unexpectedly before finishing this task.
        class_name: RandomKiller
        actor_id: 53851ad5fe3aacb063641c9201000000
        pid: 1533
        namespace: serve_failure_test
        ip: 172.31.205.150
```

I reproduced the issue locally by manually killing the actor via terminal. This caused the same error.

File: release/long_running_tests/workloads/serve_failure.py (hunk 1 above)
Comment (shrekris-anyscale) [reply release/long_running_tests/workloads/serve_failure.py:175]: It doesn't affect the test's behavior. I did it based on [@edoakes's feedback](https://github.com/ray-project/ray/pull/31945#discussion_r1088008801) from the previous change.

**Comment by cadedaniel** [comment]
Looks like relevant tests are passing! Can we merge so the weekly run this weekend will include this?

**Comment by architkulkarni** [comment]
Windows test failures unrelated

**Pull request merged**